google/cloud/bigquery/v2/minimal/internal/json_utils.cc

```diff
@@ -85,6 +85,32 @@ void ToJson(std::chrono::system_clock::time_point const& field,
   j[name] = std::to_string(m);
 }
 
+nlohmann::json RemoveJsonKeysAndEmptyFields(std::string const& json_payload,
+                                            std::vector<std::string> keys) {
+  nlohmann::json::parser_callback_t remove_empty_call_back =
+      [&keys](int depth, nlohmann::json::parse_event_t event,
+              nlohmann::json& parsed) {
+        // Have this redundant check due to compile error that
+        // depth parameter is not being used.
+        if (depth < 0) return false;
+
+        if (event == nlohmann::json::parse_event_t::key) {
+          auto const key_found = std::any_of(
+              keys.begin(), keys.end(),
+              [&parsed](std::string const& key) { return (parsed == key); });
+          return !key_found;
+        }
+        if (event == nlohmann::json::parse_event_t::object_end) {
+          if (parsed == nullptr || parsed.empty()) return false;
+        }
+        if (event == nlohmann::json::parse_event_t::array_end) {
+          if (parsed.empty()) return false;
+        }
```

Comment:
It would be clearer to always return from each event-type block, rather than allowing any fall-through.

So, in the `object_end` case, for example, ...
```
    return parsed != nullptr && !parsed.empty();
```